EXPO/screens/HomeScreen.js

```diff
@@ -1,188 +1,676 @@
+      stylers: [
+        {
+          color: "#dfd2ae"
+        }
+      ]
+    },
+    {
+      featureType: "water",
+      elementType: "geometry.fill",
+      stylers: [
+        {
+          color: "#b9d3c2"
+        }
+      ]
+    },
+    {
+      featureType: "water",
+      elementType: "labels.text.fill",
+      stylers: [
+        {
+          color: "#92998d"
+        }
+      ]
+    }
+  ];
+  componentDidMount() {
+    Geocoder.init("AIzaSyD7JZmztK5wE-80P8t-_IOHZQinVtx4Dio");
+  }
+ 
+  callNearByPlacesAPI(location) {
+   
+    const fullURL = APIConst.baseURL+JSON.stringify(location.coords.latitude)+','+JSON.stringify(location.coords.longitude)+APIConst.URNConst.nearByURN+APIConst.apiKey
+    console.log("FInal URL- "+fullURL);
+    RequestManager.requestGET(fullURL).then(res => res.json()).then(data => {
+       
+        console.log("API Response :: "+JSON.stringify(data.results));
+        this.setState({ 
+          location: location,
+          hotelsList: data.results,
+         });
+         //console.log(APIConst.basePhotoURL+data.results[0].photos.photo_reference+APIConst.photoURN+APIConst.apiKey);
+      })
+      .catch(function(error) {
+        console.log(
+          "There has been a problem with your fetch operation: " + error
+        );
+        throw error;
+      });
   }
 
-  _maybeRenderDevelopmentModeWarning() {
-    if (__DEV__) {
-      const learnMoreButton = (
-        <Text onPress={this._handleLearnMorePress} style={styles.helpLinkText}>
-          Learn more
-        </Text>
-      );
+    pinPointSearchedLocation(location) {
+      console.log("Pinporint location"+location);
+    this.setState({
+      latLng: {
+        latitude: location.latitude,
+        longitude: location.longitude
+      }
+      });
+    }
 
-      return (
-        <Text style={styles.developmentModeText}>
-          Development mode is enabled, your app will be slower but you can use useful development
-          tools. {learnMoreButton}
-        </Text>
-      );
+    calculateDistanceBetLatAndLong(lat1, lon1, lat2, lon2, unit) {
+      if ((lat1 == lat2) && (lon1 == lon2)) {
+        return 0;
+      }
+      else {
+        var radlat1 = Math.PI * lat1/180;
+        var radlat2 = Math.PI * lat2/180;
+        var theta = lon1-lon2;
+        var radtheta = Math.PI * theta/180;
+        var dist = Math.sin(radlat1) * Math.sin(radlat2) + Math.cos(radlat1) * Math.cos(radlat2) * Math.cos(radtheta);
+        if (dist > 1) {
+          dist = 1;
+        }
+        dist = Math.acos(dist);
+        dist = dist * 180/Math.PI;
+        dist = dist * 60 * 1.1515;
+        if (unit=="K") { dist = dist * 1.609344 }
+        if (unit=="N") { dist = dist * 0.8684 }
+        return dist;
+      }
+    }
+
+  componentWillMount() {
+    ScreenOrientation.allowAsync(ScreenOrientation.Orientation.ALL);
+    if (Platform.OS === "android" && !Constants.isDevice) {
+      console.log("Permission denied");
+      this.setState({
```

Comment:
we can not do setState on component which is not mounted yet, use componentDidMount instead